src/core/modules/SolarSystem.cpp

```diff
@@ -2668,14 +2611,120 @@ void SolarSystem::setNumberIsolatedTrails(int n)
 	emit numberIsolatedTrailsChanged(numberIsolatedTrails);
 }
 
+void SolarSystem::setFlagOrbits(bool b)
+{
+	if(b!=getFlagOrbits())
+	{
+		flagOrbits = b;
+		emit flagOrbitsChanged(b);
+	}
+}
+
+// Connect this to all signals when orbit selection or selected object has changed.
+// This method goes through all planets and sets orbit drawing as configured by several flags
+void SolarSystem::reconfigureOrbits()
+{
+	// we have: flagOrbits O, flagIsolatedOrbits I, flagPlanetsOrbitsOnly P, flagPermanentOrbits and a possibly selected planet S
+	// permanentOrbits only influences local drawing of a single planet and can be ignored here.
+	// O S I P
+	// 0 X X X   NONE
+	// 1 0 1 X   NONE
+	// 1 X 0 0   ALL
+	// 1 X 0 1   all planets only
+
+	// 1 1 1 0   only selected planet and orbits of its moon system
+	// 1 1 1 1   only selected SSO if it is a major planet
+
+	if (!flagOrbits || (flagOrbits&&flagIsolatedOrbits&&(!selected || selected==sun)))
+	{
+		for (const auto& p : qAsConst(systemPlanets))
+			p->setFlagOrbits(false);
+		return;
+	}
+	// from here, flagOrbits is certainly on
+	if (!flagIsolatedOrbits)
+	{
+		for (const auto& p : qAsConst(systemPlanets))
+			if (!flagPlanetsOrbitsOnly || (flagPlanetsOrbitsOnly && (p->getPlanetType()==Planet::isPlanet || (p->parent && p->parent->getPlanetType()==Planet::isPlanet) )))
```

Comment:
`!flagPlanetsOrbitsOnly || (flagPlanetsOrbitsOnly &&...)` is redundant: the second check for the flag always yields `true`. After you remove it, you can also get rid of two pairs of parentheses, yielding this simpler condition:
```
if (!flagPlanetsOrbitsOnly || p->getPlanetType()==Planet::isPlanet ||
    (p->parent && p->parent->getPlanetType()==Planet::isPlanet))
```

Reply:
Yes, indeed. This is currently written mostly documentarily.

Reply:
My concern here is that it makes it harder to understand the condition, not only because of an extra check, but also due to multiple nested levels of parentheses.